Food Sales Predictions Project

John Lynn

Part 1

In [361]:
import pandas as pd
filename = '/content/sales_predictions.csv'

In [362]:
sales_predictions_df = pd.read_csv(filename)
sales_predictions_df.head()

Item_Identifier  Item_Weight  ...        Outlet_Type  Item_Outlet_Sales
0           FDA15         9.30  ...  Supermarket Type1          3735.1380
1           DRC01         5.92  ...  Supermarket Type2           443.4228
2           FDN15        17.50  ...  Supermarket Type1          2097.2700
3           FDX07        19.20  ...      Grocery Store           732.3800
4           NCD19         8.93  ...  Supermarket Type1           994.7052

[5 rows x 12 columns]


Part 2

In [363]:
sales_predictions_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                7060 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                6113 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB


In [364]:
# Search for duplicate data
sales_predictions_df.duplicated().any()

False

Before I get into dealing with the null values, I want to clean up any inconsistencies in the columns

In [365]:
sales_predictions_df[['Item_Fat_Content','Item_Type','Outlet_Size','Outlet_Location_Type','Outlet_Type']].value_counts()

Item_Fat_Content  Item_Type              Outlet_Size  Outlet_Location_Type  Outlet_Type      
Low Fat           Household              High         Tier 3                Supermarket Type1    99
                                         Small        Tier 1                Supermarket Type1    94
                                         Medium       Tier 3                Supermarket Type3    93
                                                                            Supermarket Type2    89
                                         Small        Tier 2                Supermarket Type1    89
                                         Medium       Tier 1                Supermarket Type1    88
                  Snack Foods            Small        Tier 2                Supermarket Type1    83
                                         Medium       Tier 1                Supermarket Type1    78
                                                      Tier 3                Supermarket Type3    77
      

In [366]:
# The Item_Fat_Content column has differences in the labeling, so I need to make them consistent
sales_predictions_df['Item_Fat_Content'].replace({'LF': 'Low Fat', 'low fat': 'Low Fat', 'reg': 'Regular'}, inplace=True)

In [367]:
# Running this again to make sure the labels were changed
sales_predictions_df[['Item_Fat_Content','Item_Type','Outlet_Size','Outlet_Location_Type','Outlet_Type']].value_counts()

Item_Fat_Content  Item_Type              Outlet_Size  Outlet_Location_Type  Outlet_Type      
Low Fat           Household              High         Tier 3                Supermarket Type1    103
                                         Small        Tier 1                Supermarket Type1    103
                                                      Tier 2                Supermarket Type1    102
                                         Medium       Tier 3                Supermarket Type3     99
                                                                            Supermarket Type2     95
                                                      Tier 1                Supermarket Type1     95
                  Snack Foods            Small        Tier 2                Supermarket Type1     89
                                         Medium       Tier 3                Supermarket Type3     82
                                                      Tier 1                Supermarket Type1     

In [368]:
# Clarify the missing values
sales_predictions_df.isnull().sum()

Item_Identifier                 0
Item_Weight                  1463
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                  2410
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales               0
dtype: int64

In [369]:
# Isolate all the rows missing data for the Outlet_Size column
sales_predictions_df.loc[sales_predictions_df['Outlet_Size'].isna()]

Item_Identifier  Item_Weight  ...        Outlet_Type  Item_Outlet_Sales
3              FDX07       19.200  ...      Grocery Store           732.3800
8              FDH17       16.200  ...  Supermarket Type1          1076.5986
9              FDU28       19.200  ...  Supermarket Type1          4710.5350
25             NCD06       13.000  ...  Supermarket Type1           838.9080
28             FDE51        5.925  ...      Grocery Store           178.4344
30             FDV38       19.250  ...      Grocery Store           163.7868
33             FDO23       17.850  ...  Supermarket Type1          2174.5028
45             FDM39        6.420  ...      Grocery Store           358.2004
46             NCP05       19.600  ...  Supermarket Type1          2428.8384
47             FDV49       10.000  ...  Supermarket Type1          5815.0972
53             FDA43       10.895  ...  Supermarket Type1          3121.2704
54             NCP18       12.150  ...  Supermarket Type1          4815.0656
56             NCX54        9.195  ...  Supermarket Type1          2117.2440
61             FDV27        7.970  ...  Supermarket Type1          1062.6168
65             FDC46       17.700  ...      Grocery Store           184.4266
67             FDZ03       13.650  ...  Supermarket Type1          1118.5440
68             DRH37       17.600  ...  Supermarket Type1          2302.3364
72             FDH35       18.250  ...  Supermarket Type1          4604.6728
73             FDG02        7.855  ...  Supermarket Type1          2285.0256
78             FDL04       19.000  ...  Supermarket Type1          1587.9330
79             FDV25        5.905  ...  Supermarket Type1          5305.0944
85             FDG20       15.500  ...  Supermarket Type1          2479.4392
90             FDW20       20.750  ...      Grocery Store           369.5190
92             DRI25       19.600  ...  Supermarket Type1          1381.5350
100            FDT28       13.300  ...  Supermarket Type1          1805.6496
107            FDG12        6.635  ...  Supermarket Type1          2530.7058
111            FDS52        8.890  ...  Supermarket Type1          2732.4432
114            FDV11        9.100  ...  Supermarket Type1          3151.8972
116            FDZ16       16.850  ...  Supermarket Type1          4843.6950
118            DRA12       11.600  ...  Supermarket Type1          2552.6772
120            NCF19       13.000  ...  Supermarket Type1           680.4476
121            DRJ13       12.650  ...  Supermarket Type1          2085.9514
122            FDB14       20.250  ...      Grocery Store           186.4240
124            FDW11       12.600  ...  Supermarket Type1           866.8716
127            FDV38       19.250  ...  Supermarket Type1           928.1252
129            NCP30       20.500  ...  Supermarket Type1           707.0796
133            FDS52        8.890  ...      Grocery Store           101.2016
137            FDB51        6.920  ...  Supermarket Type1          1314.2892
138            FDX44        9.300  ...  Supermarket Type1           535.3032
139            NCN07       18.500  ...      Grocery Store           263.6568
141            NCX54        9.195  ...  Supermarket Type1          1693.7952
142            NCO55       12.800  ...  Supermarket Type1          2143.8760
150            DRJ25       14.600  ...  Supermarket Type1          1034.6532
157            FDN13       18.600  ...  Supermarket Type1          1910.1802
158            DRD24       13.850  ...  Supermarket Type1          1701.7848
165            DRL01       19.500  ...  Supermarket Type1          5375.0034
166            FDP25       15.200  ...  Supermarket Type1          2838.9712
171            FDK44       16.600  ...  Supermarket Type1          3823.0236
174            FDI32       17.700  ...      Grocery Store           345.5502
175            FDR47       17.850  ...      Grocery Store           585.2382
176            FDB35       12.300  ...  Supermarket Type1          1010.6844
1

All of the rows with null values in the Outlet_Size colum have either a Tier 2 Supermarket Type 1 or Tier 3 Grocery Store  Outlet_Location_Type and Outlet_Type

All non-null Tier 2 Supermarket Type 1 and all Grocery Store instances in the DataFrame have an Outlet_Size of Small, so I am going to replace all of the null values in the 'Outlet_Size' column with 'Small'

In [370]:
# Replace nulls with Small
sales_predictions_df['Outlet_Size'].fillna(value='Small', inplace=True)

In [371]:
# Run the .isnull again to make sure the null values of the Outlet_Size column are 0
sales_predictions_df.isnull().sum()

Item_Identifier                 0
Item_Weight                  1463
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                     0
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales               0
dtype: int64

The 1463 null values in the Item_Weight column make up 17% of that column.

I think dropping that percentage of rows from the dataset might negatively impact the rest of the data in the other columns.

I am going to get the mean of the Item_Weight column and fill the null values with that average value.

In [372]:
# First running the .mean for the entire column, including the null values
sales_predictions_df['Item_Weight'].mean()

12.857645184136183

In [373]:
# Use the .fillna to impute the column average over the null values
sales_predictions_df.fillna(12.857645184136183, inplace = True)
# Run the .info again to check the Non-Null Count again
sales_predictions_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                8523 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                8523 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB


In [374]:
# Run the .mean for the column again to make sure it looks right
sales_predictions_df['Item_Weight'].mean()

12.857645184136409

Summary Statistics of Numerical Columns

In [375]:
# Minimums
sales_predictions_df[['Item_Weight', 'Item_Visibility', 'Item_MRP', 'Outlet_Establishment_Year', 'Item_Outlet_Sales']].min()

Item_Weight                     4.555
Item_Visibility                 0.000
Item_MRP                       31.290
Outlet_Establishment_Year    1985.000
Item_Outlet_Sales              33.290
dtype: float64

In [376]:
# Maximums
sales_predictions_df[['Item_Weight', 'Item_Visibility', 'Item_MRP', 'Outlet_Establishment_Year', 'Item_Outlet_Sales']].max()

Item_Weight                     21.350000
Item_Visibility                  0.328391
Item_MRP                       266.888400
Outlet_Establishment_Year     2009.000000
Item_Outlet_Sales            13086.964800
dtype: float64

In [377]:
# Mean
sales_predictions_df[['Item_Weight', 'Item_Visibility', 'Item_MRP', 'Outlet_Establishment_Year', 'Item_Outlet_Sales']].mean()

Item_Weight                    12.857645
Item_Visibility                 0.066132
Item_MRP                      140.992782
Outlet_Establishment_Year    1997.831867
Item_Outlet_Sales            2181.288914
dtype: float64